src/test/java/com/googlecode/lanterna/issue/Issue453.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of lanterna (https://github.com/mabe02/lanterna).
+ *
+ * lanterna is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright (C) 2010-2020 Martin Berglund
+ */
+package com.googlecode.lanterna.issue;
+
+import com.googlecode.lanterna.SGR;
+import com.googlecode.lanterna.graphics.SimpleTheme;
+import com.googlecode.lanterna.graphics.TextGraphics;
+import com.googlecode.lanterna.gui2.DefaultWindowManager;
+import com.googlecode.lanterna.gui2.EmptySpace;
+import com.googlecode.lanterna.gui2.MultiWindowTextGUI;
+import com.googlecode.lanterna.TestTerminalFactory;
+import com.googlecode.lanterna.TextColor;
+import com.googlecode.lanterna.screen.Screen;
+import java.io.IOException;
+
+class Issue453 {
+
+    public static void main(String[] args) throws InterruptedException, IOException {
+        LanternaTerminalWriter writer = new LanternaTerminalWriter(args);
+        for (int i = 0; i < 1000; i++) {
+            writer.write(String.valueOf(i), SGR.BOLD);
+            Thread.sleep(100);
+        }
+        writer.close();
+    }
+
+    public static class LanternaTerminalWriter {
+
+        private Screen screen;
+        private TextGraphics screenWriter;
+
+        public LanternaTerminalWriter(String[] args) throws IOException {
+            screen = new TestTerminalFactory(args).createScreen();
+            screen.startScreen();
+            MultiWindowTextGUI gui = new MultiWindowTextGUI(screen, new DefaultWindowManager(),
+    				new EmptySpace(TextColor.ANSI.BLACK));
+            gui.setTheme(SimpleTheme.makeTheme(true, TextColor.ANSI.WHITE, TextColor.ANSI.BLACK, TextColor.ANSI.WHITE, TextColor.ANSI.WHITE, TextColor.ANSI.WHITE, TextColor.ANSI.WHITE, TextColor.ANSI.WHITE));
```

Comment:
Yes, having a Testclass for it is a good thing(TM) ;-)
I'm just surprised it doesn't test the "no-SGR" case.

Reply:
Okay nice :)
It's supposed to test the "no-SGR" case, although kind of indirectly. 
```
gui.setTheme(SimpleTheme.makeTheme(true, TextColor.ANSI.WHITE, TextColor.ANSI.BLACK, TextColor.ANSI.WHITE, TextColor.ANSI.WHITE, TextColor.ANSI.WHITE, TextColor.ANSI.WHITE, TextColor.ANSI.WHITE));
``` 
Calling the above is how I initially ran into this bug.

SimpleTheme.makeTheme [creates a SimpleTheme with no SGR](https://github.com/mabe02/lanterna/blob/50855d3ed70cc9ae9778c448633d8864fcc13345/src/main/java/com/googlecode/lanterna/graphics/SimpleTheme.java#L61) which then passes the empty array of SGR to a [new DefaultMutableThemeStyle](https://github.com/mabe02/lanterna/blob/50855d3ed70cc9ae9778c448633d8864fcc13345/src/main/java/com/googlecode/lanterna/graphics/SimpleTheme.java#L110).

I can make a more explicit test case if desired however.

Reply:
For a test, it probably should have made sure that the Theme indeed has no SGRs set, mainly because that would have immediately shown the allOf/noneOf mixup ;-)   Tests are supposed to show past bugs, even if the bug is now fixed before the PR gets merged.

Reply:
Okay good call :) I fixed it, should be better now. Let me know what you think.